components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -497,60 +514,64 @@ func SetContainerImageFromRegistry(ctx context.Context, config *rest.Config, not
 							return fmt.Errorf("invalid image selection format")
 						}
 
-						// Specify the namespaces to search in
-						namespaces := []string{"opendatahub", "redhat-ods-applications"}
+						// Specify the namespace to search in
+                        // Currently and in foreseeable future, imagestream namespace is equal to central namespace that the odh notebook controller is running in
+                        namespace, err := getControllerNamespace()
+                        if err != nil {
+                            log.Error("Get controller namespace error", "error", err)
+                            return nil
+                        }
+                        log.Info("Controller is running in namespace", "namespace", namespace)
 						imagestreamFound := false
-						for _, namespace := range namespaces {
-							// List imagestreams in the specified namespace
-							imagestreams, err := dynamicClient.Resource(ims).Namespace(namespace).List(ctx, metav1.ListOptions{})
-							if err != nil {
-								log.Info("Cannot list imagestreams", "error", err)
-								continue
-							}
+						// List imagestreams in the specified namespace
```

Comment:
weird indentation, we should run more ci to fail on this, if we dont' yet; maybe a pre-commit?

Reply:
my bad, tabs ... fixing it. About ci: yeah, would be good, possibly even pre-commit as part of linting.
Let me see what is already there. Cause I did not lint ...